src/apprt/gtk/App.zig

```diff
@@ -796,6 +797,31 @@ fn toggleQuickTerminal(self: *App) !bool {
     return true;
 }
 
+fn toggleBackgroundOpacity(
+    self: *App,
+    target: apprt.Target,
+) !void {
+    switch (target) {
+        .app => {},
+        .surface => |v| {
+            const window = v.rt_surface.container.window() orelse {
+                log.info(
+                    "toggleBackgroundOpacity invalid for container={s}",
+                    .{@tagName(v.rt_surface.container)},
+                );
+                return;
+            };
+
+            const opacity = try window.toggleBackgroundOpacity();
+
+            // Cycle through all surfaces and set the background opacity so they stay in sync
+            for (self.core_app.surfaces.items) |surface| {
+                surface.core_surface.setBackgroundOpacity(opacity);
+            }
```

Comment:
Wait what? That's not right. It should only affect the core surface of the current apprt surface

Reply:
Wouldn't we want these changes reflected across the app? If I have several tabs open and I toggle the opacity wouldn't the expected behavior be to toggle it for them all?

Reply:
Then this should be an app-level action instead of a surface-level one

Reply:
would we still want to loop through the surfaces then? Or are you suggesting avoiding that?

Reply:
Do we, in fact, want `toggle_background_opacity` to be an app-level toggle, rather than a per surface? If we had a per surface toggle, you can always make it an "app" toggle by prefixing the keybind will `all:` but there's no way to do the opposite.

Reply:
Or, must this affect all surfaces in a window?

Reply:
I mean it makes the most sense for me for this to be a "window-level" action, but since we've rejected (I think?) that for core that means we have to stick with either app or surface level for now.

Reply:
yeah i think it makes most sense to have the toggle affect only one surface. if users want different they can add `all:` to their keybind.

Reply:
Okay, so in the case of the `all` flag, we'd iterate through the surfaces as we do now; otherwise, we'd change the current surface?

Reply:
Having the `all:` flag should mean the the Ghostty core will iterate though all of the surfaces for you, we only need to handle the targeted surface.

Reply:
One caveat I noticed here: since the changes are only handled at a surface level, it's pretty easy to get a mismatch of tabs. For example, if I have 2 tabs open and I use the toggle with the `all` flag, it will toggle those two open tabs to 100 opacity. Then, if I open a new tab, it will be at my default opacity and will work opposite to the two other tabs since the state is part of the derived config at the surface level.